server/repository/src/migrations/v1_01_01/remove_sqlite_check/invoice.rs

```diff
@@ -0,0 +1,98 @@
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+        ALTER TABLE invoice ADD COLUMN status_temp NOT NULL DEFAULT 'NEW';
+        ALTER TABLE invoice ADD COLUMN type_temp NOT NULL DEFAULT 'OUTBOUND_SHIPMENT';
+
+        UPDATE invoice SET status_temp = status;
```

Comment:
I think this might need to be `UPDATE invoice SET status_temp = status WHERE status is not null`

Similar for the other fields.

Example

```
CREATE TABLE test_table (
    id TEXT,
    status TEXT
)
INSERT INTO test_table (id) VALUES ('A');
INSERT INTO test_table (id, status) VALUES ('A', 'NEW');
ALTER TABLE test_table ADD COLUMN status_temp NOT NULL DEFAULT 'NEW';
UPDATE test_table SET status_temp = status;
```
Returns
```
Execution finished with errors.
Result: NOT NULL constraint failed: test_table.status_temp
```
in sqlite at least

Reply:
I guess arguable this can't happen due to previous check constraints?
Also thought it might be worth considering...
`ALTER TABLE invoice ADD COLUMN status_temp NOT NULL DEFAULT status;`
Which again seems to work in sqlite if there are no nulls in the field.

I haven't tried what happens if you then remove the column you are defaulting from. It could break...

Reply:
> I think this might need to be UPDATE invoice SET status_temp = status WHERE status is not null

Existing status field has NOT NULL constraing

Reply:
> ALTER TABLE invoice ADD COLUMN status_temp NOT NULL DEFAULT status;

Ohh that looks cool, but if status column is dropped ?

Reply:
Lol I should real your full message before replying, basically i say the same thing as your second comment

Reply:
All good, I also tried it as a suggestion, then thought of that issue afterwards :)